### PR TITLE
:bug: UpdateObject function should operate on a deepcopy of the object

### PR DESCRIPTION
**Pull request by Jaakko-Os**
Fixes: #46

### CODE REVIEW

**Comment by maelk** [comment]
It should be here https://github.com/metal3-io/cluster-api-provider-metal3/blob/master/baremetal/metal3machine_manager.go#L1071 and here https://github.com/metal3-io/cluster-api-provider-metal3/blob/master/baremetal/metal3machine_manager.go#L1079

File: baremetal/metal3machine_manager.go [hunk 1]
```diff
@@ -207,10 +207,9 @@ func (m *MachineManager) SetPauseAnnotation(ctx context.Context) error {
 		if _, ok := annotations[bmh.PausedAnnotation]; ok {
 			m.Log.Info("BaremetalHost is already paused")
 			return nil
-		} else {
-			m.Log.Info("Adding PausedAnnotation in BareMetalHost")
-			host.Annotations[bmh.PausedAnnotation] = pausedAnnotationKey
 		}
+		m.Log.Info("Adding PausedAnnotation in BareMetalHost")
+		host.Annotations[bmh.PausedAnnotation] = pausedAnnotationKey
```
Comment (maelk) [inline baremetal/metal3machine_manager.go]: why changing this ? Kashif did it this way so that we would not override the user's pause annotation

Comment (Jaakko-Os) [reply baremetal/metal3machine_manager.go]: lint complaint fix, the 'else' has no effect since the 'if' has a 'return'

Comment (maelk) [reply baremetal/metal3machine_manager.go]: ah ok then fine by me

**Pull request merged**